Flick/Controllers/LoginViewController.swift

```diff
@@ -49,41 +104,129 @@ class LoginViewController: UIViewController {
                        let profileImagePngData = profileImage.pngData() {
                         base64Str = profileImagePngData.base64EncodedString()
                     }
-
-                    NetworkManager.authenticateUser(
-                        username: "",
-                        firstName: firstName,
-                        lastName: lastName,
-                        profilePic: base64Str,
-                        socialId: profile.userID,
-                        socialIdToken: accessToken) { [weak self] authorizationToken in
-                        guard let self = self else { return }
-                        DispatchQueue.main.async {
-                            self.userDefaults.set(authorizationToken, forKey: Constants.UserDefaults.authorizationToken)
-                            self.navigationController?.pushViewController(HomeViewController(), animated: true)
-                        }
-                    }
                 }
+
+                self.authenticateUser(
+                    firstName: firstName,
+                    lastName: lastName,
+                    email: email,
+                    profilePic: base64Str,
+                    socialId: profile.userID,
+                    socialIdToken: accessToken,
+                    socialIdTokenType: "facebook"
+                )
             }
         }
     }
 
-    override func viewWillAppear(_ animated: Bool) {
-        super.viewWillAppear(animated)
-        navigationController?.setNavigationBarHidden(true, animated: true)
+    private func authenticateUser(firstName: String,
+                                  lastName: String,
+                                  email: String?,
+                                  profilePic: String,
+                                  socialId: String,
+                                  socialIdToken: String,
+                                  socialIdTokenType: String) {
+        NetworkManager.authenticateUser(
+            firstName: firstName,
+            lastName: lastName,
+            email: email,
+            profilePic: profilePic,
+            socialId: socialId,
+            socialIdToken: socialIdToken,
+            socialIdTokenType: socialIdTokenType) { [weak self] authorizationToken in
+            guard let self = self else { return }
+            DispatchQueue.main.async {
+                self.userDefaults.set(authorizationToken, forKey: Constants.UserDefaults.authorizationToken)
+                let homeViewController = HomeViewController()
+                self.navigationController?.pushViewController(homeViewController, animated: true)
+            }
+        }
+    }
+
+}
+
+extension LoginViewController: ASAuthorizationControllerDelegate, ASAuthorizationControllerPresentationContextProviding {
+
+    func presentationAnchor(for controller: ASAuthorizationController) -> ASPresentationAnchor {
+        return self.view.window!
```

Comment:
is this safe?

Reply:
i think it should be, it's what's used in apple's documentation